secure_message/api_mocks/internal_user_service_mock.py

```diff
@@ -7,14 +7,50 @@
 
 class InternalUserServiceMock:
     def __init__(self):
-        self.internal_user_dict = {"Someuuid": {"id": "Someuuid",
-                                                "firstName": "fred",
-                                                "lastName": "flinstone",
-                                                "emailAddress": "mock@email.com"}}
+        self.internal_user_dict = {
+            "Someuuid": {
+                "id": "Someuuid",
+                "firstName": "fred",
+                "lastName": "flinstone",
+                "emailAddress": "mock@email.com"
+            },
+            "01b51fcc-ed43-4cdb-ad1c-450f9986859b": {
+                "id": "01b51fcc-ed43-4cdb-ad1c-450f9986859b",
+                "firstName": "fred",
+                "lastName": "flinstone",
+                "emailAddress": "mock@email.com"
+            },
+            "f62dfda8-73b0-4e0e-97cf-1b06327a6712": {
+                "id": "01b51fcc-ed43-4cdb-ad1c-450f9986859b",
+                "firstName": "fred",
+                "lastName": "flinstone",
+                "emailAddress": "mock@email.com"
+            },
+            "Tej": {
+                "id": "Tej",
+                "firstName": "Tejas",
+                "lastName": "patel",
+                "emailAddress": "mock@email.com"
+            },
+            "BRES": {
+                "id": "TRES",
+                "firstName": "BRES",
+                "lastName": "",
+                "emailAddress": "mock@email.com"
+            },
+            "SomeStringUntilWeGetIds": {
+                "id": "SomeStringUntilWeGetIds",
+                "firstName": "MadeUpUser",
+                "lastName": "",
+                "emailAddress": "mock@email.com"
+            }
+        }
 
     def get_user_details(self, uuid):
         """gets the user details from the internal user service"""
         logger.debug("getting mock user details for uaa")
         if uuid:
-            return self.internal_user_dict[uuid], 200
-        return "error retrieving details"
+            found = self.internal_user_dict.get(uuid)
+            if found:
+                return found, 200
```

Comment:
I would have done like : 
if  uuid in elf.internal_user_dict
  return true, 200

return False, 400 

just for consistency in the return type. and log with error/warning user not found.

Reply:
Can we try and use EAFP instead of LBYL? https://blogs.msdn.microsoft.com/pythonengineering/2016/06/29/idiomatic-python-eafp-versus-lbyl/

```Python
try:
    return self.internal_user_dict[uuid], 200
except KeyError:
    logger.exception(f'No internal user with uuid {uuid}'
    return "error retrieving details", 404
```

Reply:
Also, don't inspect the uuid arg. If it's None then we'll still catch the exception, and if we set None as a key in a dictionary we have bigger problems.

Reply:
Normally I would support EAFP instead of LBYL . However in this case the existing case and party mocks are written so as to mimic the http response which would not raise an exception to the code  but return an error . In this specific of the internal_user_service_mock it is something of a guess to assume it will be a http response . If it isnt and we do get exceptions I will change the mock implementation to suit. The uuid in elf.internal_user_dict needs to look in the values()

Reply:
My suggestion doesn't change the return values, only the way in which the decision for what values to return is made

Reply:
ok , misunderstood , changed